tools/converter/mule2ballerina/src/main/java/org/wso2/ei/tools/mule2ballerina/configreader/ConfigReader.java

```diff
@@ -0,0 +1,224 @@
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.ei.tools.mule2ballerina.configreader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.ei.tools.mule2ballerina.elementmapper.AttributeMapper;
+import org.wso2.ei.tools.mule2ballerina.elementmapper.ElementMapper;
+import org.wso2.ei.tools.mule2ballerina.model.BaseObject;
+import org.wso2.ei.tools.mule2ballerina.model.Flow;
+import org.wso2.ei.tools.mule2ballerina.model.GlobalConfiguration;
+import org.wso2.ei.tools.mule2ballerina.model.Processor;
+import org.wso2.ei.tools.mule2ballerina.model.Root;
+import org.wso2.ei.tools.mule2ballerina.util.Constant;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.EndElement;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * {@code ConfigReader} class reads mule configuration file and builds the Mule tree
+ */
+public class ConfigReader {
+
+    private static Logger logger = LoggerFactory.getLogger(ConfigReader.class);
+
+    private ElementMapper mapperObject;
+    private AttributeMapper attributeMapper;
+    private Root mRoot;
+    private boolean flowStarted = false;
+    private List<String> unIdentifiedElements;
+
+    public ConfigReader() {
+        mapperObject = new ElementMapper();
+        attributeMapper = new AttributeMapper();
+        mRoot = new Root();
+        unIdentifiedElements = new ArrayList<String>();
+    }
+
+    public void readXML(InputStream inputStream) {
+
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+        try {
+            XMLEventReader eventReader = xmlInputFactory.createXMLEventReader(inputStream);
+
+            while (eventReader.hasNext()) {
+                XMLEvent xmlEvent = eventReader.nextEvent();
+                switch (xmlEvent.getEventType()) {
+                case XMLStreamConstants.START_ELEMENT:
```

Comment:
Formatting of switch statement

Reply:
Done. Pull request for that can be viewed here >> https://github.com/wso2/product-ei/pull/1216